lib/sync/goals.ts

```diff
@@ -52,10 +53,11 @@ export function isSyncRepoCommit(sdm: SoftwareDeliveryMachine): PushTest | undef
     }
     return pushTest("IsSyncRepoCommit", async p => {
         const repo: SyncRepoRef | RemoteRepoRef = _.get(sdm, "configuration.sdm.k8s.options.sync.repo");
-        if (!isRemoteRepo(repo)) {
+        if (!isRemoteRepo(repo) && !isInLocalMode()) {
             throw new Error(`SyncRepoRef did not get converted to proper RemoteRepoRef at startup: ${stringify(repo)}`);
         }
-        if (p.id.providerType === repo.providerType && p.id.owner === repo.owner &&
+        if (isInLocalMode() || (p.id.providerType === (repo as RemoteRepoRef).providerType) &&
```

Comment:
I needed to add the `isInLocalMode()` checks otherwise the the pushTest would always fail (when running in local mode)